distributed/scheduler.py

```diff
@@ -1426,11 +1426,16 @@ def handle_worker(self, worker):
                     for msg in msgs:
                         if msg == 'OK':  # from close
                             break
-
                         if 'status' in msg and 'error' in msg['status']:
-                            logger.error("error from worker %s: %s",
+                            try:
+                                logger.error("error from worker %s: %s",
                                          worker, clean_exception(**msg)[1])
-
```

Comment:
Why catch `ImportError`'s specifically?  Are there not other cases where deserializing the exception might cause an exception?  In fact, we might be able to test this by creating an exception that can be serialized, but fails on deserialization.

```python
class UnDeserializableException(Exception):
    def __getstate__(self):
        return 
    def __setstate__(self, state):
        raise Exception('Some unique text to search for in a test')
```

Reply:
You are right. Unpickling can generate a variety of Exceptions. I'll broaden it.